### PR TITLE
Mark injected properties in Presenters as initialized

### PR DESCRIPTION
**Pull request by rattuscz**
Any idea how to create test?
I've tried to look at phpstan-doctrine but that does not seem to have test for this type of extension (or I did not find it)

### CODE REVIEW

File: src/Rule/Nette/PresenterInjectedPropertiesExtension.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+{
+
+	public function isAlwaysRead(PropertyReflection $property, string $propertyName): bool
+	{
+		return false;
+	}
+
+	public function isAlwaysWritten(PropertyReflection $property, string $propertyName): bool
+	{
+		return false;
```
Comment (ondrejmirtes) [inline src/Rule/Nette/PresenterInjectedPropertiesExtension.php:18]: The property is written, you could call `return $this->isInitialized()` for simplification.

File: src/Rule/Nette/PresenterInjectedPropertiesExtension.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	}
+
+	public function isInitialized(PropertyReflection $property, string $propertyName): bool
+	{
+		return $property->isPublic() &&
+			$property->getDeclaringClass()->implementsInterface('Nette\Application\IPresenter') &&
```
Comment (ondrejmirtes) [inline src/Rule/Nette/PresenterInjectedPropertiesExtension.php:24]: This isn't just for presenters. You can skip this condition.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
You could create something like this: https://github.com/phpstan/phpstan-strict-rules/tree/master/tests/Levels

Do not create the json files - write the test files, re-run the tests a few times until they're green, the json files are created automatically.

File: src/Rule/Nette/PresenterInjectedPropertiesExtension.php (hunk 2 above)
Comment (rattuscz) [reply src/Rule/Nette/PresenterInjectedPropertiesExtension.php:24]: oki, wrong `@inject` could hide misconfiguration error (class not processed by DI) but to find it the whole DI config would need to be analyzed.. NetteStan :-D

Comment (ondrejmirtes) [reply src/Rule/Nette/PresenterInjectedPropertiesExtension.php]: I don't think this is currently a concern, we can simply consider all properties with `@inject` being initialized, it's a better state of things than what it currently reports.

But asking for a DI container is not unprecedented - phpstan-doctrine and phpstan-symfony do something similar. We can do that in the next stage if you feel like it.

**Comment by rattuscz** [comment]
ok finally added test based on the levels example

**Comment by rattuscz** [comment]
Travis umira na : `defined constant XDEBUG_CC_UNUSED - assumed 'XDEBUG_CC_UNUSED' (this will throw an Error in a future version of PHP)`
jedine pouziti vidim v phpunit code-coverage driveru, nejaky BC break zavislosti?

**Comment by ondrejmirtes** [comment]
@rattuscz No idea, please try to turn off Xdebug in Travis.

**Comment by ondrejmirtes** [comment]
I love it, thank you! Gonna release it ASAP.

**Comment by ondrejmirtes** [comment]
Tagged as 0.12.11.

**Comment by rattuscz** [comment]
@ondrejmirtes ouch, the test ends up green but actually it fails to generate code coverage, i was playing on my cloned repo on different branch but could not make it work.
But the problem is not in the changes, just that travis now includes xdebug 3.0 which is incompatible with the phpunit version you use.
And it boils down to undefined constant, maybe that's all required to "fix it" :-)

**Comment by ondrejmirtes** [comment]
@rattuscz I don't care much about code coverage, the support for generating that can be removed...

**Pull request merged**